sbeidas/django-vue-dockerized-celery
# Ali

## Taks Description
The goal of this exercise is to create an API that calls both the Github and Bitbucket APIs to load the most recent commits from two public repositories and returns the commits to the user. An example from Github is:
- https://github.com/pytest-dev/pytest
The corresponding API endpoint:
- https://api.github.com/repos/pytest-dev/pytest/commits
Before you start:
- Fork this repo:
    - https://github.com/sbeidas/django-vue-dockerized-celery
- This is a sample skeleton app for you to use, please checkout the README to get an idea about the tools being used
- Run the Skeleton APP locally
    - docker-compose up --build
- Ping the user API http://0.0.0.0:8000/api/users (you can open this in your browser), this returns the list of all users in the database. 
1) When the API is hit, it should ping the upstream APIs and get the commits 
2) To trigger the import of the commits, please make sure to do so asynchronously, for your convenience Celery (a python async library) has been built in the code base
    - To test out celery, SSH into one of the boxes (make sure they are running first)
        - docker exec -it django-vue-dockerized-celery_worker_1 sh
        - python manage.py shell
    - In python then you can run:
        - from apps.users.tasks import add
        - add.delay(x=2, y=2)
    - from the logs notice how the `add` job ran on the `worker` docker instance 
Example Bonus Tasks:
If you are done with task above, here are some suggested bonuses to work on:
- Sync stauses:
  [] Expose and API to show the user if a sync task is in progress
- API Improvemens
  [] Add a search functionality to search by commit hash
  [] Return statuses pertaining to a commit
  [] Add pagination for the commits
  [] Add filtering by author
